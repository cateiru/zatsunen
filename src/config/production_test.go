package config_test

import (
	"testing"

	"github.com/cateiru/zatsunen/src/config"
	"github.com/gkampitakis/go-snaps/snaps"
	"github.com/stretchr/testify/require"
)

func TestProductionConfig(t *testing.T) {
	production := config.SetProductionConfig("/path")

	snaps.MatchSnapshot(t, production)

	require.Equal(t, production.GetMode(), "production")
	snaps.MatchSnapshot(t, production.GetConfig())
}
