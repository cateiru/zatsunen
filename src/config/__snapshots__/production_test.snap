
[TestProductionConfig - 1]
config.Local{
    Config: config.Config{
        Mode:         "production",
        CommonConfig: config.CommonConfig{},
        LogConfig:    config.LogConfig{
            Options: &slog.HandlerOptions{
                AddSource:   true,
                Level:       slog.Level(0),
                ReplaceAttr: func([]string, slog.Attr) slog.Attr {...},
            },
            IsJsonLog: true,
        },
        MiddlewareConfig: config.MiddlewareConfig{
            AllowSecFetchSiteValues: {"same-origin", "same-site"},
        },
    },
}
---

[TestProductionConfig - 2]
config.Config{
    Mode:         "production",
    CommonConfig: config.CommonConfig{},
    LogConfig:    config.LogConfig{
        Options: &slog.HandlerOptions{
            AddSource:   true,
            Level:       slog.Level(0),
            ReplaceAttr: func([]string, slog.Attr) slog.Attr {...},
        },
        IsJsonLog: true,
    },
    MiddlewareConfig: config.MiddlewareConfig{
        AllowSecFetchSiteValues: {"same-origin", "same-site"},
    },
}
---
